tests/test_gitlab.py

```diff
@@ -11,3 +19,23 @@ def test_parse_no_edition(self):
     def test_is_ee(self):
         assert gitlab.Version.parse('9.4.0-ee').is_ee
         assert not gitlab.Version.parse('9.4.0').is_ee
+
+
+class TestApiCalls(unittest.TestCase):
+    def test_success_immediately_no_response(self):
+        api = gitlab.Api(HTTPBIN, "", append_api_version=False)
+        self.assertTrue(api.call(GET("/status/202")))
+        self.assertTrue(api.call(GET("/status/204")))
+        self.assertFalse(api.call(GET("/status/304")))
+
+    def test_failure_after_all_retries(self):
+        api = gitlab.Api(HTTPBIN, "", append_api_version=False)
+
+        with self.assertRaises(gitlab.Conflict):
+            api.call(GET("/status/409"))
+
+        with self.assertRaises(gitlab.TooManyRequests):
+            api.call(GET("/status/429"))
+
+        with self.assertRaises(gitlab.GatewayTimeout):
+            api.call(GET("/status/504"))
```

Comment:
Side note, but we should enable GitHub Actions also now to ensure stuff like this gets tested. I looked at them and they are safe apart from the `deploy` workflow which I created a PR to remove now: https://github.com/hiboxsystems/marge-bot/pull/7

Reply:
@slovdahl - if you amend the commit here and push it once more, the CI job should run. It'll likely fail just like in https://github.com/hiboxsystems/marge-bot/actions/runs/3747379517/jobs/6363544706 but I can give that a look in a separate PR.

Reply:
Done :heavy_check_mark: